core/primitives/tipset/tipset.cpp

```diff
@@ -147,7 +153,22 @@ namespace fc::primitives::tipset {
     return outcome::success();
   }
 
-  outcome::result<Randomness> Tipset::randomness(
+  outcome::result<Randomness> Tipset::beaconRandomness(
+      Ipld &ipld,
+      DomainSeparationTag tag,
+      ChainEpoch round,
+      gsl::span<const uint8_t> entropy) const {
+    auto ts{this};
```

Comment:
why?

Reply:
Need to store current tipset across iterations.
On first iteration (of when loop has no iterations) pointer refers to "this". On subsequent iterations it refers to local variable "parent" moved from "loadParent".
This way there is no unnecessary copy of "this".